Sources/Signal.swift

```diff
@@ -571,6 +571,31 @@ extension SignalProtocol {
 			}
 		}
 	}
+	
+	/// Applies `transform` to values from `signal` and forwards values with non `nil` results unwrapped.
+	/// - parameters:
+	///   - transform: A closure that accepts a value from the `value` event and
+	///                returns a new optional value.
+	///
+	/// - returns: A signal that will send new values, that are non `nil` after the transformation.
+	public func filterMap<U>(_ transform: @escaping (Value) -> U?) -> Signal<U, Error> {
+		return Signal { observer in
```

Comment:
Every `Signal` means a couple more calls and locking, though we are gradually slashing a microsecond here and there.

It is not like a complex algorithm that does benefit from reusing implementations anyway...